### PR TITLE
feat: comment search

### PR DESCRIPTION
**Pull request by c0rydoras**
implements: https://github.com/adfinis/timed/issues/29

### CODE REVIEW

**Comment by winged** [comment]
Quick question: how will this be used? As you note yourself, this may cause performance problems, not only for malicious queries (which I think we can neglect here) but also for regular users who don't know when a RE query will cause the DB to break down

**Comment by winged** [comment]
> @winged #29 describes that

Thanks! Still a small explanation in the PR (and commit message!) won't hurt, along with a link in the original comment to the issue. Just to make things easier when we need to dig for reasons in 5 years

**Comment by c0rydoras** [comment]
@winged i think we could use the `search` [lookup](https://docs.djangoproject.com/en/5.0/ref/contrib/postgres/search/)

**Comment by c0rydoras** [comment]
I switched to postgres full text search instead of regex, this should make it so that ReDoS is no longer an issue, and it's probably more user friendly for nontechnical users :)

**Review by open-dynaMIX — CHANGES_REQUESTED** [review]
Backend change looks good :+1:

However, it would be cleaner if this PR just contained two commits:

 - feat: added full text search for comments
 - ci(frontend): use pnpm v9

File: frontend/app/components/task-selection/component.js
Comment (open-dynaMIX) [inline frontend/app/components/task-selection/component.js:53]: Still mentioning `regex`.

File: frontend/app/components/task-selection/component.js
Comment (open-dynaMIX) [inline frontend/app/components/task-selection/component.js:228]: Same here.

**Review by open-dynaMIX — CHANGES_REQUESTED** [review]
Well done, with the commit cleanup :+1:

I found two references to `regex` in the changes.
After that, we just need a frontend approval :tada:

**Comment by winged** [comment]
I much prefer the FTS search! However this will need an appropriate index on the `comment` field, otherwise it's gonna be rather slow, as it needs a full table scan each time, same as the regex lookup would have.

Here's how you can do that: https://docs.djangoproject.com/en/5.0/ref/contrib/postgres/search/#performance

File: frontend/app/components/task-selection/template.hbs
Comment (derrabauke) [inline frontend/app/components/task-selection/template.hbs:57]: I think the `comment-input` should be part of the `filter-sidebar` component, instead of the `task-selection`. It's more of a *filter* and the `task-selection` is used in various places. Also makes the whole initialization stuff easier, as you don't have to struggle with the `task-selection` component (which is *horrifying IMHO*)... :)

Comment (c0rydoras) [reply frontend/app/components/task-selection/template.hbs]: i moved it to `filter-sidebar/filter` https://github.com/adfinis/timed/pull/283/commits/89f2bf1eaf68795e16bc54fe3b22ead5d1e14d15#diff-cdc879d2c92851bda53f78bd96507f73ea47da51b6f9c084e65bc0ca4fe09d48

Comment (derrabauke) [reply frontend/app/components/task-selection/template.hbs]: Sorry for being so picky here, but I think this splintered component (previously `comment-input`) is not well maintainable. :sweat_smile: 

I wanted to suggest decoupling the whole comment search input from the `task-selection` component. Just like you started out in the beginning, with your `comment-input`, but instead of loading it into the task-selection component why not directly use it in the `analytics/index` template like we do for the [user select](https://github.com/adfinis/timed/commit/89f2bf1eaf68795e16bc54fe3b22ead5d1e14d15#diff-12c469691b8d9581a646f849b70514cb098e3442b584edecf96d2b6f9ac75079R60)? 

The `task-selection` component should really not become any more bloated as it already is.

So:
* (re-)add `comment-input` or whatever you wanna call it
* make direct use of it in `analysis/index` template
* keeps `filters-sidebar/filter` and `task-selection` untouched --> less :spaghetti: more :sparkles:

Comment (c0rydoras) [reply frontend/app/components/task-selection/template.hbs]: yea i agree, i changed it now

**Review by derrabauke — APPROVED** [review]

**Review by open-dynaMIX — APPROVED** [review]

**Pull request merged**
